controllers/suite_test.go

```diff
@@ -45,14 +47,93 @@ const (
 )
 
 var (
-	testEnv           *helpers.TestEnvironment
-	clusterReconciler *ClusterReconciler
-	ctx               = context.Background()
+	testEnv  *helpers.TestEnvironment
+	ctx      = context.Background()
+	testEnv1 *helpers.TestEnvironment
+	ctx1     = context.Background()
 )
 
+func TestMain(m *testing.M) {
+	fmt.Println("Creating new test environment")
+	testEnv1 = helpers.NewTestEnvironment()
+
+	// Set up a ClusterCacheTracker and ClusterCacheReconciler to provide to controllers
+	// requiring a connection to a remote cluster
+	tracker, err := remote.NewClusterCacheTracker(
+		log.Log,
+		testEnv1.Manager,
+	)
+	if err != nil {
+		panic(fmt.Sprintf("unable to create cluster cache tracker: %v", err))
+	}
+	if err := (&remote.ClusterCacheReconciler{
+		Client:  testEnv1,
+		Log:     log.Log,
+		Tracker: tracker,
+	}).SetupWithManager(testEnv1.Manager, controller.Options{MaxConcurrentReconciles: 1}); err != nil {
+		panic(fmt.Sprintf("Failed to start ClusterCacheReconciler: %v", err))
+	}
+	// TODO: (wfernandes) Do we need these reconcilers for the MHC tests?
```

Comment:
I believe these reconcilers can be added if we convert the remaining ginkgo/envtest tests to use the vanilla go test pattern. I've commented these out so that we can focus on the bare minimum for what's needed for the MHC tests.

Reply:
The way some of these tests are written, I don't see how these are "unit" tests anymore. They definitely feel like integration tests since they rely on other reconcilers to set state on certain objects like `Machine` and `MachineSet` in order for the MHC reconciler to work.